### ISSUE TITLE
📝 HTML: inconsistent closing-slash on void elements

### ISSUE THREAD

**Issue by echocrow** [biomejs/biome#6656]
### Description

when using `"selfCloseVoidElements": "never"` (default), seems that the forward slash is not consistently stripped ATM.
the two bugs I'm seeing:

given: `<meta foo="bar" />`
```diff
expected..actual
-<meta foo="bar">
+<meta foo="bar" />
```

given: `<meta foo="bar">`
```diff
expected..actual
-<meta foo="bar">
+<meta foo="bar" >
```

oddly enough, this issue seems to only be present on single-line void elements.
when the element has a long list of attributes (forcing multi-line formatting), the trailing slash is correctly stripped.\*

\* **update:** looks like formatting of multi-line void elements has its own little bug: currently those may need to be formatted _twice_ to yield the expected result.

specifically, given a long void element starting on a single line and closing with `/>`, the first format split it into multiple lines and keeps the `/>`. then, a subsequent format removes the closing slash, closing it with just `>` as expected.

should I open a separate issue for this "requires double formatting" bug?

### Environment information

```bash
CLI:
  Version:                      2.0.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.7.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Formatter:
  Format with errors:           unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Attribute position:           unset
  Bracket spacing:              unset
  Includes:                     unset

JavaScript Formatter:
  Enabled:                      unset
  JSX quote style:              unset
  Quote properties:             unset
  Trailing commas:              unset
  Semicolons:                   unset
  Arrow parentheses:            unset
  Bracket spacing:              unset
  Bracket same line:            unset
  Quote style:                  unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Attribute position:           unset

JSON Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Trailing Commas:              unset
  Expand lists:                 unset

CSS Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Quote style:                  unset

GraphQL Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Bracket spacing:              unset
  Quote style:                  unset

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
  "$schema": "https://biomejs.dev/schemas/2.0.6/schema.json",
  "html": {
    "formatter": {
      "enabled": true
    }
  }
}
```

### Playground link

https://biomejs.dev/playground/?code=PAAhAGQAbwBjAHQAeQBwAGUAIABoAHQAbQBsAD4ACgA8AGgAdABtAGwAIABsAGEAbgBnAD0AIgBlAG4AIgA%2BAAoACQA8AGgAZQBhAGQAPgAKAAkACQA8AG0AZQB0AGEAIABmAG8AbwA9ACIAYgBhAHIAIgA%2BAAoACQAJADwAbQBlAHQAYQAgAGYAbwBvAD0AIgBiAGEAcgAiACAALwA%2BAAoACgAJAAkAPABtAGUAdABhAAoACQAJAAkAZgBvAG8APQAiAGIAYQByACIACgAJAAkACQBjAG8AbgB0AGUAbgB0AD0AIgB2AGUAcgB5ACAAbABvAG4AZwAgAGEAdAB0AHIAaQBiAHUAdABlAHMAIAB0AG8AIABmAG8AcgBjAGUAIABtAHUAbAB0AGkALQBsAGkAbgBlACAAZgBvAHIAbQBhAHQAdABpAG4AZwAiAAoACQAJAD4ACgAJAAkAPABtAGUAdABhAAoACQAJAAkAZgBvAG8APQAiAGIAYQByACIACgAJAAkACQBjAG8AbgB0AGUAbgB0AD0AIgB2AGUAcgB5ACAAbABvAG4AZwAgAGEAdAB0AHIAaQBiAHUAdABlAHMAIAB0AG8AIABmAG8AcgBjAGUAIABtAHUAbAB0AGkALQBsAGkAbgBlACAAZgBvAHIAbQBhAHQAdABpAG4AZwAiAAoACQAJAC8APgAKACAAIAAgACAAIAAgAAoACQAJADwAbQBlAHQAYQAgAGYAbwBvAD0AIgBiAGEAcgAiACAAYwBvAG4AdABlAG4AdAA9ACIAdgBlAHIAeQAgAGwAbwBuAGcAIABhAHQAdAByAGkAYgB1AHQAZQBzACAAdABvACAAZgBvAHIAYwBlACAAbQB1AGwAdABpAC0AbABpAG4AZQAgAGYAbwByAG0AYQB0AHQAaQBuAGcAIgAgAC8APgAKAAkAPAAvAGgAZQBhAGQAPgAKAAoACQA8AGIAbwBkAHkAPgAKAAkACQA8AGgAMQA%2BAGgAZQBsAGwAbwAgAHcAbwByAGwAZAA8AC8AaAAxAD4ACgAJADwALwBiAG8AZAB5AD4ACgA8AC8AaAB0AG0AbAA%2BAAoA&language=html

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
This isn't a bug. The option is called `selfCloseVoidElements` for a reason: it's only applied to **void elements**. Void elements are the following: https://html.spec.whatwg.org/#void-elements

[area](https://html.spec.whatwg.org/#the-area-element), [base](https://html.spec.whatwg.org/#the-base-element), [br](https://html.spec.whatwg.org/#the-br-element), [col](https://html.spec.whatwg.org/#the-col-element), [embed](https://html.spec.whatwg.org/#the-embed-element), [hr](https://html.spec.whatwg.org/#the-hr-element), [img](https://html.spec.whatwg.org/#the-img-element), [input](https://html.spec.whatwg.org/#the-input-element), [link](https://html.spec.whatwg.org/#the-link-element), [meta](https://html.spec.whatwg.org/#the-meta-element), [source](https://html.spec.whatwg.org/#the-source-element), [track](https://html.spec.whatwg.org/#the-track-element), [wbr](https://html.spec.whatwg.org/#the-wbr-element)

It's possible that we made the feature very specific, and maybe you're looking to remove the self-closing tag from all tags?
